VideoLocker/src/main/java/org/edx/mobile/module/notification/EdxParsePushBroadcastReceiver.java

```diff
@@ -32,101 +33,129 @@
-                courseBundle.putBoolean(Router.EXTRA_ANNOUNCEMENTS, true);
-                courseBundle.putString(Router.EXTRA_COURSE_ID, payload.getCourseId());
-
-
-                Intent resultIntent = new Intent(context, CourseDetailTabActivity.class);
-                resultIntent.setAction(Long.toString(System.currentTimeMillis()));
-                //if user launch the app from recent list, activity will still get this intent.
-                //this is one way to avoid it.
-                resultIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK |Intent.FLAG_ACTIVITY_EXCLUDE_FROM_RECENTS);
-                resultIntent.putExtra( Router.EXTRA_BUNDLE, courseBundle);
-
-                // Because clicking the notification opens a new ("special") activity, there's
-                // no need to create an artificial back stack.
-                PendingIntent resultPendingIntent =  PendingIntent.getActivity(
-                        context,
-                        0,
-                        resultIntent,
-                        PendingIntent.FLAG_UPDATE_CURRENT
-                );
-
-                NotificationManager notificationManager =
-                        (NotificationManager) context.getSystemService(Context.NOTIFICATION_SERVICE);
-
                 NotificationCompat.Builder builder = new NotificationCompat.Builder(context);
                 builder.setContentTitle(titleTemplate);
                 builder.setContentText(payload.getCourseName());
                 builder.setSmallIcon(R.drawable.app_icon);
-                builder.setContentIntent(resultPendingIntent);
                 builder.setAutoCancel(true);
 
-                Uri uri= RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
+                Uri uri = RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION);
                 builder.setSound(uri);
 
+                setNotificationIntents(context, intent, builder);
+
                 Notification notification = builder.build();
                 notification.flags |= Notification.FLAG_AUTO_CANCEL;
                 return notification;
+
             } else {
                 EventBus.getDefault().postSticky(new CourseAnnouncementEvent(
-                    CourseAnnouncementEvent.EventType.MESSAGE_RECEIVED, ""));
-                return super.getNotification(context,intent);
+                        CourseAnnouncementEvent.EventType.MESSAGE_RECEIVED, ""));
+                return super.getNotification(context, intent);
             }
 
         } catch (Exception e) {
             return super.getNotification(context, intent);
         }
     }
+
+    /***
+     * Code to set OPEN and DELETE intents so that onPushOpen and onPushDismiss are called.
+     * This method's contents were copied from {@link ParsePushBroadcastReceiver#getNotification(Context, Intent)}
+     */
+    private void setNotificationIntents(Context context, Intent intent, NotificationCompat.Builder builder) {
```

Comment:
Can this method be changed to create a `NotificationCompat.Builder` itself and return that, so that we can replace the builder creation with calling this method?